### ISSUE TITLE
Copy-File does not continue copying on error

### ISSUE THREAD

**Issue by pchatham** [PSAppDeployToolkit/PSAppDeployToolkit#301]
I have come across an issue where `Copy-File` does not continue copying when hitting an error such as 'file in use'

I have added -ContinueOnError $true
It does not continue. This standard PS Function `Copy-Item`, however , does work

PSAppDeployToolkit Funtion Example (Not Working) :
`Copy-File -Path "$source\Images\*" -Destination '$dest\Images\' -Recurse -ContinueOnError $true`

Standard Powershell Function Example (Working):
`Copy-Item -Path "$source\Images\*" -Destination '$dest\Images\' -Force -Recurse -ErrorAction Continue`

Looking at the function code, looks like `$ContinueOnError` is designed to just handle the error but continue the script without existing, but also stops copying files?.. where as in the PS Function, Copy-Item, `-ErrorAction Continue` would actually continue copying files.

Is this correct?

**Comment by pchatham** [comment]
if this is confirmed as working as designed then this can be closed.
A note in the Documentation might help.

**Comment by kmaiman3** [comment]
On Sep 15, 2017 05:03, pchatham <notifications@github.com> wrote:

I have come across an issue where i cannot copy files over existing files if they are in use
I have added -ContinueOnError $true
It does not continue. Copy-Item, however , does work

PSAppDeployToolkit Funtion Example (Not Working) :
Copy-File -Path "$source\Images\*" -Destination '$dest\Images\' -Recurse -ContinueOnError $true

Standard Powershell Example (Working):
Copy-Item -Path "$source\Images\*" -Destination '$dest\Images\' -Force -Recurse -ErrorAction Continue

**Comment by seanlillis** [comment]
We used ErrorAction "Stop" to terminate what would otherwise be non-terminating errors with the native Copy-Item function. The reason for this was a) to capture and log any errors, and b) to ensure the script failed if any critical file copy operations were not completed successfully. The ContinueOnError parameter allows the script to continue despite errors, but the Copy-Item function still terminates once an error is encountered, so any remaining files are not copied.

I have added a new parameter ContinueFileCopyOnError parameter to the File-Copy function for the next release. This will ensure that files continue to be copied if an error is encountered. This uses Copy-Item with ErrorAction "SilentlyContinue". Any errors are captured as warnings in the log file. I have also enhanced the logging for this function. Feel free to download the latest source files and test.
